frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -154,15 +157,25 @@ export function getParameterMappingOptions(
 
   const { isNative } = Lib.queryDisplayInfo(question.query());
   if (!isNative) {
-    const { query, stageIndex, columns } = getParameterColumns(
+    const { query, columns } = getParameterColumns(
       question,
       parameter ?? undefined,
     );
-    const columnGroups = Lib.groupColumns(columns);
 
-    const options = columnGroups.flatMap(group =>
-      buildStructuredQuerySectionOptions(query, stageIndex, group),
-    );
+    const groups = _.uniq(columns, ({ group }) => group);
```

Comment:
We should not make any assumptions about CLJS entities being the same or not. If we need a unique group key, let's add it. You could use a group index.

Reply:
8847f13aac9dd5b4b72694785be6c6ab5533aea3